packages/contracts-bedrock/src/L1/OptimismPortal2.sol

```diff
@@ -760,34 +758,39 @@ contract OptimismPortal2 is Initializable, ResourceMetering, ReinitializableBase
         }
     }
 
-    /// @notice Add a minter to the OptimismPortal contract. To disable, set an empty value.
-    function setMinter(address _minter) external {
-        if (msg.sender != proxyAdminOwner()) {
-            revert OptimismPortal_Unauthorized();
-        }
-        QKCConfigStorage storage $ = _getQKCConfigStorage();
-        $.minter = _minter;
-        emit MinterSet(_minter);
-    }
-
     /// @notice Mint a specific amount of L2 native token to an address.
-    function mintTransaction(address _to, uint256 _value) external metered(RECEIVE_DEFAULT_GAS_LIMIT) {
-        QKCConfigStorage storage $ = _getQKCConfigStorage();
-        if (msg.sender != $.minter) {
+    /// @dev    This function is only callable by L1CrossDomainMessenger.
+    function mintTransaction(
```

Comment:
It appears that `mintTransaction` and `depositTransaction` share most of the same code. We could implement a shared `_depositTransaction` internal function containing the common logic, while allowing mintTransaction and depositTransaction to each handle their specific access control requirements.

Reply:
I'd like to keep `depositTransaction` untouched, otherwise it'll be a pain to merge in future.